cpp/core/operators/shuffle/splitter.cc

```diff
@@ -37,13 +41,31 @@ using arrow::internal::checked_cast;
 #define SPLIT_BUFFER_SIZE 16 * 1024 * 1024
 #endif
 
+// macro to rotate left an 8-bit value 'x' given the shift 's' is a 32-bit integer
+// (x is left shifted by 's' modulo 8) OR (x right shifted by (8 - 's' modulo 8))
+#if !defined(__x86_64__)
+#define rotateLeft(x, s) (x << (s - ((s>>3)<<3)) | x >> (8 - (s - ((s>>3)<<3))))
+#endif
+
+// on x86 machines, _MM_HINT_T0,T1,T2 are defined as 1, 2, 3
+// equivalent mapping to __builtin_prefetch hints is 3, 2, 1
+#if defined(__x86_64__)
```

Comment:
Can XSimd be a better way to hide the difference?

Reply:
We could, but since this is CPP operators code, and prefetch instructions seem to be used only here, I think this change can be localized here.